h/h_api/bulk_api/command_processor.py

```diff
@@ -104,20 +104,26 @@ def _check_command_count(self, final=False):
         :param final: This is the final count check, not incremental
         """
 
-        if not self.config:
-            return
+        total = self.config.total_instructions if self.config else None
 
-        total = self.config.total_instructions
+        if final:
+            if not self.command_count:
+                raise CommandSequenceError("No instructions received")
 
-        if not final:
-            if self.command_count > total:
+            total = self.config.total_instructions
```

Comment:
Does this duplicate the `total = self.config.total_instructions if self.config else None` above?

Reply:
Yeah, but `self.config` must be present by this point. I'll have another look.

Reply:
Nope, it's not what I thought at all. Looks like I was half way through de-duplicating this and just didn't finish.